crates/rust-analyzer/src/main_loop.rs

```diff
@@ -706,18 +717,10 @@ fn on_diagnostic_task(task: DiagnosticTask, msg_sender: &Sender<Message>, state:
     }
 }
 
-fn send_startup_progress(
-    sender: &Sender<Message>,
-    loop_state: &mut LoopState,
-    world_state: &WorldState,
-) {
-    if !world_state.feature_flags.get("notifications.workspace-loaded") {
-        return;
-    }
-
-    let total: usize = world_state.workspaces.iter().map(|it| it.n_packages()).sum();
+fn send_startup_progress(sender: &Sender<Message>, loop_state: &mut LoopState) {
+    let total: usize = loop_state.roots_total;
     let prev_progress = loop_state.roots_scanned_progress;
-    let progress = total - world_state.roots_to_scan;
+    let progress = total - loop_state.roots_to_scan;
```

Comment:
This seems a bit convoluted, maybe we could change `roots_to_scan` to `roots_scanned`?

Reply:
Good idea.